packages/fxa-content-server/app/scripts/views/settings.js

```diff
@@ -142,7 +150,17 @@ const View = BaseView.extend({
   beforeRender() {
     const account = this.getSignedInAccount();
 
-    return account.fetchProfile().then(() => this.user.setAccount(account));
+    return account
+      .fetchProfile()
+      .then(() => this.user.setAccount(account))
+      .then(() => account.settingsData())
```

Comment:
It seems like you'll need to pass the force=true parameter here, otherwise you may not detect when something's gone wrong with the subscription or credit card. Or, if force=true might hammer on the endpoint because of React re-rendering, maybe do the `settingsData` caching at this view layer, so that a hard page refresh would re-fetch the data?

Reply:
> ...so that a hard page refresh would re-fetch the data?

Oh wait, maybe I've missed something then. So the account model is persisted in local storage or somewhere? I didn't even realise.